frontend/src/components/core/Block/Block.tsx

```diff
@@ -166,27 +142,37 @@ class Block extends PureComponent<Props> {
     width: number
   ): ReactNode | null {
     const element = reportElement.get("element")
-
-    if (element.get("type") === "empty") {
-      // Just a plain div -- so we're sure the rendered element has no
-      // height/margin/padding. Also saves some CPU cycles as a side-effect.
-      return <div className="stEmpty" key={index}></div>
-    }
-
     const component = this.renderElement(
       element,
       index,
       width,
       reportElement.get("metadata")
     )
 
+    const componentWithMaybe = (
+      <Maybe
+        enable={
+          // Note that this will toggle the shouldComponentUpdate flag.
+          // shouldComponentUpdate is used on the update flow, not on the
+          // mount flow. When mounting this check is effectively skipped.
+          element.get("type") !== "empty" ||
+          this.props.reportRunState !== ReportRunState.RUNNING
+        }
+      >
+        {component}
+      </Maybe>
+    )
+
     const isStale =
-      this.props.showStaleElementIndicator &&
-      this.isElementStale(reportElement)
+      !componentWithMaybe.props.enable ||
+      (this.props.showStaleElementIndicator &&
+        this.isElementStale(reportElement))
 
     const className =
```

Comment:
i think this assignment is not very readable. I'd simplify this with something like:
After adding `!FullScreenWrapper.isFullScreen` to the `isStale` conditions we can do:
```
    const className = classNames('element-container', {
      "stale-element":  isStale,
      "stEmpty": !isStale && element.get("type") === "empty"
    })
```